### ISSUE TITLE
ONNX Model making random predictions.

### ISSUE THREAD

**Issue by mayank-nference** [microsoft/onnxruntime#8660]
**Describe the bug**
I converted the fine-tuned PyTorch NER (Named Entity Recognition) bert model to onnx format to speed up the inferencing. But for a fair number of inputs onnx model is generating random predictions. ONNX model accuracy is decreased to 60 %, whereas the original PyTorch model has 98% accuracy. Why onnx model is making such random predictions for inputs?? Do I need to re-train the onnx model to improve the accuracy? And there is no improvement in speed, though I was able to reduce the size of the model up to 4x via onnx dynamic quantization.
**Urgency**
I need to resolve this issue as soon as possible.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 20.04.2 LTS
- ONNX Runtime installed from (source or binary): pip install onnxruntime-gpu
- ONNX Runtime version: 1.8.1
- ONNX installed from (source or binary): conda install -c conda-forge onnx
- ONNX version: 1.10.1
- Python version: 3.7
- CUDA/cuDNN version: 11.3
- GPU model and memory: Model: RTX 2060 Memory: 6GB

**Expected behavior**
The original PyTorch model generates logits 1 or 2. But the onnx model generates logits 0, 1 or 2. why 0?? this should not happen.

**Comment by faxu** [comment]
A few questions to help us debug:
- Can you share the model that's resulting in incorrect predictions? 
- Which version of PyTorch did you use to export the model?  
- What steps did you use for quantization? 
- Do you see the same issues using a CPU build of ORT?

**Comment by mayank-nference** [comment]
A few questions to help us debug:

* Can you share the model that's resulting in incorrect predictions?
    * yes I have attached here - [onnx_model](https://drive.google.com/file/d/1F5Tdwpt_q0yFKUCVkPE6_GfqaLBOI22g/view?usp=sharing)
*  Which version of PyTorch did you use to export the model?
     * I have used PyTorch stable (1.9.0) 
* What steps did you use for quantization?
    * This is how I have exported my model - 
* `def convert_to_onnx(model, tokenizer):
    model, tokenizer = GetModel(OUTPUT_DIR=onnx_config.MODEL_PATH)
    in_sentence = onnx_config.SENTENCE
    model_inputs = GetModelInputs(tokenizer, model_config, onnx_config.SENTENCE)
    input_ids,input_mask,segment_ids=model_inputs["input_ids"],model_inputs["input_mask"],model_inputs["segment_ids"]
    torch.onnx.export(
        model,
        (input_ids, input_mask, segment_ids),
        "./onnx_models/model.onnx",
         export_params=True,
        opset_version=11,  
        do_constant_folding=True,
        input_names=["input_ids", "input_mask", "segment_ids"],
        output_names=["logits", "pooled_output_drop"],
        dynamic_axes={
            "input_ids" : {0 : "batch_size"},
            "input_mask" : {0 : "batch_size"},
            "segment_ids" : {0 : "batch_size"},
         }
    )` 

* Do you see the same issues using a CPU build of ORT?
    * YES

**Comment by YUNQIUGUO** [comment]
Thanks for sharing the model.

@garymm Hi, is this accuracy issue likely to be associated with the conversion? any comments?
 If yes, I can close the issue and open a new issue on pytorch-onnx repo. If not, I can tag the inference team to take a look

**Comment by garymm** [comment]
Possible it's due to conversion, but I wouldn't say "likely". Someone needs to debug the issue to find the root cause. My team is unlikely to have time to do that any time soon. If it is root-caused to a converter issue, please open an issue in pytorch/pytorch with label `module: onnx`.

**Comment by mayank-nference** [comment]
@YUNQIUGUO @garymm any updates?? How much time will it take to get this resolve? Any rough idea?

**Comment by ytaous** [comment]
Hi, @mayank-nference, 

To help us to root cause it, can u pls provide more details on reproducible steps:
1. the sample torch code and inputs that run thru the model to generate outputs (make it small and simple like unit test will be great)
2. the onnxruntime/inference code with (the same) inputs that run thru the converted model to generate outputs
3. the outputs comparison for the sample inputs above (which you might have already) shows decreasing accuracy

Thanks.
